### PR TITLE
improve(rooms): Support for joining rooms by pressing enter from UserInput keyDown prop

### PR DESCRIPTION
**Pull request by samjgorman**
## Description of Changes

- Add support for passing a `keyDown` handler for `UserInput`
- Use this in `Home` to support entering a room by pressing enter when the input UI is focused

## Testing Checklists
If your change adds critical functionality that should be tested on every PR, edit ``pull_request_template.md`` and add it to the checklist.
### Basic Author Checklist 
The PR author should verify that the following are fully functional.
- [x] With 1 user:
  - [x] The world loads, the user can move/look around, and physics behave properly
  - [x] User can pause their client, mute/unmute themselves, and change their name, with the appropriate interactions showing in the pause menu
- [x] With 2 users:
  - [x] A second user can join the same room, and both users can see each others model/video and hear each others audio
  - [x] State changes (name changes, A/V toggles) propagate between peers as expected
  - [x] Can leave a room from the UI, rejoin that room from the UI, then leave & conect from the UI to a new room

### PR Specific Author Checklist
What did you do to test your change in particular? For example, if your changes touches A/V functionality, how did you stress-test this part of the product?
- [ ] xyz

### Advanced Author Checklist
These are more advanced checks that might be helpful for some PRs. Completing these is at the discretion of the author and reviewer.
- [ ] With 2 users, memory stays within a reasonable interval over a 5 minute period
- [ ] With 2 users, the server remains healthy after an extended amount of time in the world
- [ ] With 3+ users, the server and client behave in a manner consistent with 2 peers

### Basic Reviewer Checklist
The reviewer of this PR should load SpaceBar off this branch and ensure the following work as expected:
- [ ] 2 user can join the world and see each other's model/video, and hear each other's audio
- [ ] Movement and collisions behave as expected
- [ ] Changing some state (name, microphone, video) for one user propagates correctly to the other user

### CODE REVIEW

File: src/interface/Home/Home.tsx [hunk 1]
```diff
@@ -29,6 +35,7 @@ const Home = () => {
               placeholder="Name of your space"
               maxLength={30}
```
Comment (shalins) [inline src/interface/Home/Home.tsx:36]: const?

File: src/interface/Home/Home.tsx [hunk 2]
```diff
@@ -15,6 +15,12 @@ const Home = () => {
     pushHardRefresh(`/${newRoomName}`);
   };
 
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter') {
```
Comment (shalins) [inline src/interface/Home/Home.tsx:19]: const? also do certain keyboard events ever get registered as lowercase `enter`?

**Review by shalins — APPROVED** [review]
Approving to unblock but noticed some magic numbers in the code

File: src/interface/Home/Home.tsx (hunk 2 above)
Comment (samjgorman) [reply src/interface/Home/Home.tsx:19]: Will verify but from our work writing `usePlayerControls` hook believe they're coded as uppercase

File: src/interface/Home/Home.tsx (hunk 1 above)
Comment (samjgorman) [reply src/interface/Home/Home.tsx:36]: Good idea

File: src/interface/Home/Home.tsx (hunk 2 above)
Comment (samjgorman) [reply src/interface/Home/Home.tsx:21]: See here: https://developer.mozilla.org/en-US/docs/Web/API/UI_Events/Keyboard_event_code_values

Comment (shalins) [reply src/interface/Home/Home.tsx:21]: nice! thanks for the reference, as a nit for the future, would be great to include a map going from enum => string with a comment including that link above, so future keypresses could just be queried as `KeyboardInput[KeyboardInput.Enter]` for example

Comment (samjgorman) [reply src/interface/Home/Home.tsx:21]: This is a good idea for future work!  We can then have `usePlayerControls` and this logic read from the same shared enum. As a first step towards this I'll link to this documentation in the code

Comment (samjgorman) [reply src/interface/Home/Home.tsx]: Actually decided to just bundle in this refactor in this PR -- code should be simpler + more maintainable now

**Pull request merged**